File: backend/user-auth-service/src/main/java/com/tripify/user_auth_service/dto/request/CompanionDto.java
package com.tripify.user_auth_service.dto.request;

import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.time.LocalDate;
import java.util.UUID;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class CompanionDto {
    private UUID id;
    private String firstName;
    private String lastName;

    @JsonFormat(pattern = "yyyy-MM-dd")
    private LocalDate dateOfBirth;
}

File: frontend/app/src/main/java/com/tripify/tripify_android/profile/model/CompanionDto.kt
package com.tripify.tripify_android.profile.model

data class CompanionDto(
    val id: String? = null, // Null quando lo inviamo per crearlo
    val firstName: String,
    val lastName: String,
    val dateOfBirth: String // Formato "YYYY-MM-DD"
)